ptvsd/wrapper.py

```diff
@@ -1027,6 +1032,54 @@ def on_threads(self, request, args):
 
         self.send_response(request, threads=threads)
 
+    @async_handler
+    def on_source(self, request, args):
+        """Request to get the source"""
+        source_reference = args.get('sourceReference', 0)
+        filename = '' if source_reference == 0 else \
+            self.source_map.to_pydevd(source_reference)
+
+        if source_reference == 0:
+            self.send_error_response(request, 'Source unavailable')
+        else:
+            server_filename = pydevd_file_utils.norm_file_to_server(filename)
+
+            cmd = pydevd_comm.CMD_LOAD_SOURCE
+            _, _, content = yield self.pydevd_request(cmd, server_filename)
+            self.send_response(request, content=content)
+
+    def get_source_reference(self, filename):
+        """Gets the source reference only in remote debugging scenarios.
+        And we know that the path returned is the same as the server path
+        (i.e. path has not been translated)"""
+
+        if self.start_reason == 'launch' or \
```

Comment:
I wonder if it's worth disabling it for launch, given that "remote" attach can also be local - and often is in practice (e.g. for most people using our debugger with embedded Python scripting). It looks like the code below would take care of things either way?

Reply:
Sorry, but I'm not following you here.